openapi.json

```
@@ -430,6 +430,9 @@
                 }
               }
             }
+          },
+          "plots":{
+            "$ref": "#/components/schemas/LongTermPlots"
```

```suggestion
            "$ref": "#/components/schemas/TermPlots"
```
Instead of defining different term plots like `LongTermPlots`, `MediumTermPlots` and `ShortTermPlots`. Can we DRY principle only have one like `TermPlots`.

---

I was willing to do the same however for each term number of datapoints differ as example for short term its 4, medium term its 7 and long term its 15. At the same time timestamps differ, as example for short term its every 6 hours, for medium and long term its every day. So to match API response I had to keep these separate.

---

Ok got it. In that case we can keep components `LongTermPlots`, `MediumTermPlots` and `ShortTermPlots`.
Can we change `ShortTermPlotsData`, `MediumTermPlotsData`, `LongTermPlotsData` to single component `TermPlotsData`?

---

I think no, because every `ShortTermPlotsData`, `MediumTermPlotsData` and `LongTermPlotsData` have different number of properties with varying timestamps?

---

I see only `timestamp` is different but other two `cpuUsage` and `memoryUsage` are the same. I think we can keep the same timestamp for all the examples.

---

Yeah we can, however then it does not depict exact response?

---

As per discussion with Sagnik its not needed match exact api response in openapi. With that I reduced the file, and it has far less duplicate details now.